extension/src/completion/componentCompletionItemProvider.ts

```diff
@@ -43,6 +43,22 @@ export class ComponentCompletionItemProvider implements vscode.CompletionItemPro
                 { value: "InternalMember" }
             ])
         },
+        { tag: "TransitionData", attribute: "Id", provider: new IdCompletionProvider("TransitionData") },
+        {
+            tag: "TransitionData", provider: new BasicCompletionProvider([
+                { value: "Id" },
```

Comment:
is it possible to add the timeout related fields only if the transition is of type TimeOut?

Reply:
I don't think so. The snippet mechanism is not as "clever" as we would like...
According to other extensions, the solution is to create different snippets:
in our case, xctransition and xctimeouttransition
What do you think about it ?

Reply:
lol it's exactly your proposal in your other comment ! +1

Reply:
But this code is for autocompletion, not snippet, isn't it?
That's why I proposed different snippets, and type specific autocompletion.

Reply:
Ok I understand. Currently this autocomplete just provides completion based on all available xml attributes.
If we want to add logic in this part, we can do it I think (just a more challenging regex lol)
Perhaps we can create another ticket for that. I would like to provide all the basic intellisense as soon as possible

Reply:
new ticket #96